src/js/utils/string-utils.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Useful string handling utilities.
+ */
+define(['underscore'], function(_) {
+    'use strict';
+
+    var interpolateEscape, escapeInterpolateOptions, HTML;
+
+    /**
+     * Creates an HTML snippet.
+     *
+     * @param {string} text The HTML for the text.
+     * @constructor
+     */
+    function HTMLSnippet(text) {
+        this.text = text;
+    }
+    HTMLSnippet.prototype.toString = function() {
+        return this.text;
+    };
+
+    HTML = function(text) {
+        return new HTMLSnippet(text);
+    };
+
+    /**
+     * Returns an HTML-escaped text string by substituting values into
```

Comment:
The way I have this right now, HTMLSnippet is an internal class, so I didn't want to document it. I suppose I could make it public and then the function would be typed more appropriately.